vms/evm/subscriber.go

```diff
@@ -128,75 +128,53 @@ func (s *subscriber) processBlockRange(
 func (s *subscriber) getHeaderByNumberRetryable(headerNumber *big.Int) (*types.Header, error) {
 	var err error
 	var header *types.Header
-	attempt := 1
-	for {
+	operation := func() (err error) {
 		header, err = s.rpcClient.HeaderByNumber(context.Background(), headerNumber)
-		if err == nil {
-			return header, nil
-		}
-		s.logger.Warn(
+		return err
+	}
+	err = utils.WithMaxRetries(operation, retryMaxElapsedTime, s.logger)
```

Comment:
Let's use a default rpc retry timeout specified in the utils package, that I suggested adding in another comment.